### ISSUE TITLE
Fix false positives on vue external source style tag in no-empty-source

### ISSUE THREAD

**Issue by noe132** [stylelint/stylelint#3314]
**Describe the bug? (A clear and concise description of what the bug is)**
when `no-empty-source` is on, it treats vue style tag using src for a seperated css/less file as a empty source

**Which rule, if any, is this issue related to?**
`no-empty-source`

**What CSS is needed to reproduce this issue?**
```vue
<template>
  <div> content </div>
</template>
<style lang="less" src="./style.less" scoped></style>
```

**What stylelint configuration is needed to reproduce this issue?**
```js
module.exports = {
  'rules': {
     'no-empty-source': true,
  },
}
```

**Which version of stylelint are you using?**
`9.2.1`

**How are you running stylelint: CLI, PostCSS plugin, Node API?**
CLI with `stylelint .\src\xxx\xxx.vue`"

**Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?**
I don't think so

**What did you expect to happen?**
maybe ignore for the style tag has an src attribute in vue files?

**What actually happened (e.g. what warnings or errors you are getting)?**
```shell
src/xxx/xxx.vue
4:38  ×  Unexpected empty source   no-empty-source
```

**Comment by gucong3000** [comment]
Closing this issue, because it was fixed in #3316
